### PR TITLE
Add "Signalled groups" functionality and documentation improvements

### PR DESCRIPTION
**Pull request by brunompc**

### CODE REVIEW

File: src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt [hunk 1]
```diff
@@ -119,7 +125,8 @@ class ApplicationContextListener(val assignmentRepository: AssignmentRepository,
             val assignment = Assignment(id = "sampleJavaProject", name = "Sample Java Assignment",
                     packageName = "org.dropProject.samples.sampleJavaAssignment", ownerUserId = "teacher1",
                     submissionMethod = SubmissionMethod.UPLOAD,
-                    gitRepositoryUrl = "git@github.com:palves-ulht/sampleJavaAssignment.git",
+                    //gitRepositoryUrl = "git@github.com:palves-ulht/sampleJavaAssignment.git",
+                    gitRepositoryUrl = "git@github.com:brunompc/sampleJavaAssignment.git",
```
Comment (palves-ulht) [inline src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt:129]: I don't feel confortable with this dependency. In fact, I don't feel confortable with the dependency on palves-ulht repo either. There is an "official" repo for the sampleJavaAssignment on https://github.com/drop-project-edu/sampleJavaAssignment. Can you (1) make a PR of your changes on the sampleJavaAssignment repo and (2) change the url on this line?

File: src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt [hunk 2]
```diff
@@ -155,6 +169,7 @@ class ApplicationContextListener(val assignmentRepository: AssignmentRepository,
 
             if (connected) {
 
+                /*
```
Comment (palves-ulht) [inline src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt:172]: Why is this commented?

File: src/main/kotlin/org/dropProject/controllers/ReportController.kt [hunk 3]
```diff
@@ -95,22 +99,63 @@ class ReportController(
...
+    @RequestMapping(value = ["/cenas/{assignmentId}"], method = [(RequestMethod.GET)])
+    fun getSignaledGroupsOrSubmissions(@PathVariable assignmentId: String, model: ModelMap,
+    principal: Principal, request: HttpServletRequest): String {
+        model["assignmentId"] = assignmentId
+
+        assignmentService.getAllSubmissionsForAssignment(assignmentId, principal, model, request,
+                includeTestDetails = true,
+                mode = "signalledSubmissions")
+
+        return "signalledSubmissions"
```
Comment (palves-ulht) [inline src/main/kotlin/org/dropProject/controllers/ReportController.kt:120]: I know I'm being nit-picky but can you change this template's name to "signalled-submissions", to be coherent with the naming scheme used by the other templates?

File: src/main/kotlin/org/dropProject/data/GroupedProjectGroups.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+package org.dropProject.data
+
+import org.apache.coyote.http11.Constants.a
```
Comment (palves-ulht) [inline src/main/kotlin/org/dropProject/data/GroupedProjectGroups.kt:22]: Is this import supposed to be here?

File: src/main/kotlin/org/dropProject/services/AssignmentService.kt [hunk 5]
```diff
@@ -19,28 +19,44 @@
...
+        val assigneeRepository: AssigneeRepository,
+        val submissionService: SubmissionService,
+        val assignmentTestMethodRepository: AssignmentTestMethodRepository
 ) {
 
     @Cacheable(
             value = ["archivedAssignmentsCache"],
             key = "#principal.name",
             condition = "#archived==true")
+
```
Comment (palves-ulht) [inline src/main/kotlin/org/dropProject/services/AssignmentService.kt:51]: Please switch the documentation with the annotation (see https://stackoverflow.com/questions/3109950/codestyle-put-javadoc-before-or-after-annotation)

File: src/main/resources/templates/report.html [hunk 6]
```diff
@@ -27,6 +27,9 @@ <h1 class="page-header" th:text="${assignmentId} + ' submissions'"></h1>
         <li role="presentation" th:classappend="${mode == 'testMatrix'} ? 'active' : ''">
             <a th:href="@{'/testMatrix/' + ${assignmentId}}" data-toggle="tab" data-target="#myTab">Test Matrix</a>
         </li>
+        <li role="presentation" th:classappend="${mode == 'signalledSubmissions'} ? 'active' : ''">
+            <a th:href="@{'/cenas/' + ${assignmentId}}" data-toggle="tab" data-target="#myTab">Signalled Students</a>
```
Comment (palves-ulht) [inline src/main/resources/templates/report.html:31]: Please change "cenas" to "signalledSubmissions" or something... :)

File: src/main/resources/templates/test-matrix.html [hunk 7]
```diff
@@ -27,6 +27,9 @@ <h1 class="page-header" th:text="${assignmentId} + ' submissions'"></h1>
         <li role="presentation" th:classappend="${mode == 'testMatrix'} ? 'active' : ''">
             <a th:href="@{'/testMatrix/' + ${assignmentId}}" data-toggle="tab" data-target="#myTab">Test Matrix</a>
         </li>
+        <li role="presentation" th:classappend="${mode == 'signalledSubmissions'} ? 'active' : ''">
+            <a th:href="@{'/cenas/' + ${assignmentId}}" data-toggle="tab" data-target="#myTab">Signalled Students</a>
```
Comment (palves-ulht) [inline src/main/resources/templates/test-matrix.html:31]: same comment about "cenas"

**Review by palves-ulht — CHANGES_REQUESTED** [review]
Great stuff! Left some comments with minor suggestions.

File: src/main/kotlin/org/dropProject/services/AssignmentService.kt (hunk 5 above)
Comment (brunompc) [reply src/main/kotlin/org/dropProject/services/AssignmentService.kt]: Done

File: src/main/kotlin/org/dropProject/controllers/ReportController.kt (hunk 3 above)
Comment (brunompc) [reply src/main/kotlin/org/dropProject/controllers/ReportController.kt]: Done.

File: src/main/resources/templates/report.html (hunk 6 above)
Comment (brunompc) [reply src/main/resources/templates/report.html]: Done

File: src/main/resources/templates/test-matrix.html (hunk 7 above)
Comment (brunompc) [reply src/main/resources/templates/test-matrix.html]: Done

File: src/main/kotlin/org/dropProject/data/GroupedProjectGroups.kt (hunk 4 above)
Comment (brunompc) [reply src/main/kotlin/org/dropProject/data/GroupedProjectGroups.kt]: Not really. Removed.

File: src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt (hunk 2 above)
Comment (brunompc) [reply src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt]: As agreed, I will uncomment this code and do a new PR

File: src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt (hunk 1 above)
Comment (brunompc) [reply src/main/kotlin/org/dropProject/controllers/ApplicationContextListener.kt]: Done.

**Review by palves-ulht — APPROVED** [review]

**Pull request merged**